module_utils/network_lsr/argument_validator.py

```diff
@@ -2097,6 +2097,18 @@ def _validate_post(self, value, name, result):
                             "a infiniband device with 'infiniband.p_key' "
                             "property also needs 'mac' or 'parent' property",
                         )
+                    if "parent" in result:
+                        if "interface_name" in result:
+                            raise ValidationError(
+                                name + ".interface_name",
+                                "the 'interface_name' does not need to be specified when "
+                                "'infiniband.p_key' is not '-1' and the 'parent' exists",
+                            )
```

Comment:
This error is only ok, if InfiniBand support was broken in the past and there are no playbooks out there that would fail now.

Reply:
Yes, the ipoib support was broken before, so it is safe to raise error here.